### PR TITLE
CSP-6 Add tests to service and model

### PR DESCRIPTION
**Pull request by DevenKShah**
I had to refactor certain bits based on the data that we are going to build for pilot.

### CODE REVIEW

File: src/SFA.DAS.Roatp.Api.UnitTests/Services/GetProviderCoursesServiceTests/GetAllCoursesTests.cs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        [Test]
+        public async Task GetAllCourses_ProviderFound_ReturnsListOfCourseModel()
+        {
+            _mockProviderReadRepository.Setup(p => p.GetByUkprn(It.IsAny<int>())).ReturnsAsync(new Provider());
+
+            _mockCourseRepository.Setup(m => m.GetAllProviderCourses(It.IsAny<int>())).ReturnsAsync(_courses);
+
+            var sut = new GetProviderCoursesService(_mockCourseRepository.Object, _mockProviderReadRepository.Object);
+
+            var courses = await sut.GetAllCourses(1);
```
Comment (MarkFCain) [inline src/SFA.DAS.Roatp.Api.UnitTests/Services/GetProviderCoursesServiceTests/GetAllCoursesTests.cs:48]: can you turn the 1 into a named variable please

File: src/SFA.DAS.Roatp.Api.UnitTests/Services/GetProviderCoursesServiceTests/GetAllCoursesTests.cs [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+            Assert.AreEqual(2, courses.Count);
+        }
+
+        [Test]
+        public async Task GetAllCourses_ProviderNotFound_ReturnsEmptyList()
+        {
+            var sut = new GetProviderCoursesService(_mockCourseRepository.Object, _mockProviderReadRepository.Object);
+
+            var courses = await sut.GetAllCourses(1);
```
Comment (MarkFCain) [inline src/SFA.DAS.Roatp.Api.UnitTests/Services/GetProviderCoursesServiceTests/GetAllCoursesTests.cs:58]: see previous note

File: src/SFA.DAS.Roatp.Api.UnitTests/Services/GetProviderCoursesServiceTests/GetCourseTests.cs [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        {
+            _mockProviderReadRepository.Setup(p => p.GetByUkprn(It.IsAny<int>())).ReturnsAsync(new Provider());
+
+            var course = new ProviderCourse
+            {
+                LarsCode = 1,
+                IfateReferenceNumber = "ST1001"
+            };
+
+            _mockCoursesRepo.Setup(m => m.GetProviderCourse(It.IsAny<int>(), It.IsAny<int>())).ReturnsAsync(course);
```
Comment (MarkFCain) [inline src/SFA.DAS.Roatp.Api.UnitTests/Services/GetProviderCoursesServiceTests/GetCourseTests.cs:36]: can you change the second parameter in the setup into 1 (to match the lars code), and change the sut.GetCourse to use this value as the second parameter?

Comment (MarkFCain) [reply src/SFA.DAS.Roatp.Api.UnitTests/Services/GetProviderCoursesServiceTests/GetCourseTests.cs:36]: Also turn the '1' into a named variable, easier to see what's going on

File: src/SFA.DAS.Roatp.Api.UnitTests/Services/GetProviderCoursesServiceTests/GetAllCoursesTests.cs [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+            {
+                LarsCode = 1,
+                IfateReferenceNumber = "ST1001"
+            },
+            new ProviderCourse
+            {
+                LarsCode = 2,
+                IfateReferenceNumber = "ST1002"
+            }
+        };
```
Comment (raviudari01) [inline src/SFA.DAS.Roatp.Api.UnitTests/Services/GetProviderCoursesServiceTests/GetAllCoursesTests.cs:26]: It is good to have test data in a separate test fixture class

Comment (DevenKShah) [reply src/SFA.DAS.Roatp.Api.UnitTests/Services/GetProviderCoursesServiceTests/GetAllCoursesTests.cs:26]: I would consider this when I write more complex/acceptance tests. I like self contained tests so it is easy to navigate/change things.

File: src/SFA.DAS.Roatp.Api.UnitTests/Services/GetProviderCoursesServiceTests/GetAllCoursesTests.cs (hunk 1 above)
Comment (DevenKShah) [reply src/SFA.DAS.Roatp.Api.UnitTests/Services/GetProviderCoursesServiceTests/GetAllCoursesTests.cs:48]: Instead I have used named parameter to make it more readable.

**Review by raviudari01 — APPROVED** [review]

**Pull request merged**